plugins/node/opentelemetry-instrumentation-mongodb/src/instrumentation.ts

```diff
@@ -288,6 +324,15 @@ export class MongoDBInstrumentation extends InstrumentationBase {
     };
   }
 
+  private _getV4ConnectionPoolCheckOut() {
+    return (original: V4ConnectionPool['checkOut']) => {
+      return function patchedCheckout(this: unknown, callback: any) {
+        const patchedCallback = AsyncResource.bind(callback);
```

Comment:
It might be helpful for future maintenance to have a comment here that this patch will be unnecessary for future versions of mongodb that include the https://github.com/mongodb/node-mongodb-native/pull/3871 fix.

Reply:
done!